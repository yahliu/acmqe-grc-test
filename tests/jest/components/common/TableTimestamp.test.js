/* Copyright (c) 2020 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'

import renderer from 'react-test-renderer'
import React from 'react'
import TableTimestamp from '../../../../src-web/components/common/TableTimestamp'

Date.now = jest.fn(() => 1600291034000)

describe('PolicyTemplateDetailsView component', () => {
  it('renders as expected (recent)', () => {
    const component = renderer.create(
      <TableTimestamp
        timestamp={'2020-09-14T18:32:33Z'}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('renders as expected (long ago)', () => {
    const component = renderer.create(
      <TableTimestamp
        timestamp={'2015-09-14T18:32:33Z'}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('renders as expected -- unknown timestamp', () => {
    const component = renderer.create(
      <TableTimestamp />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
