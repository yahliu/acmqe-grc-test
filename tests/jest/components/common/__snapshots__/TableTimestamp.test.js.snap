// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyTemplateDetailsView component renders as expected (long ago) 1`] = `
<span>
  5 years ago
</span>
`;

exports[`PolicyTemplateDetailsView component renders as expected (recent) 1`] = `
<span>
  2 days ago
</span>
`;

exports[`PolicyTemplateDetailsView component renders as expected -- unknown timestamp 1`] = `
<span>
  unknown
</span>
`;
